3rdparty/fastcv/src/fastcv_hal_imgproc.cpp

```diff
@@ -992,4 +992,60 @@ int fastcv_hal_cvtBGRtoYUVApprox(
 
     fcvStatus status = FASTCV_SUCCESS;
     CV_HAL_RETURN(status, hal_BGRtoYUVApprox);
+}
+
+int fastcv_hal_canny(
+    const uchar*    src_data,
+    size_t          src_step,
+    uchar*          dst_data,
+    size_t          dst_step,
+    int             width,
+    int             height,
+    int             cn,
+    double          lowThreshold,
+    double          highThreshold,
+    int             ksize,
+    bool            L2gradient)
+{
+    int numThreads = cv::getNumThreads();
+
+    if(numThreads!=1)
+        CV_HAL_RETURN_NOT_IMPLEMENTED("API performs optimally in single-threaded mode");
+
+    if (cn != 1)
+        CV_HAL_RETURN_NOT_IMPLEMENTED("Multi-channel input is not supported");
+
+    if (lowThreshold > highThreshold)
+        CV_HAL_RETURN_NOT_IMPLEMENTED("lowThreshold is greater then highThreshold");
+
+    double	lowThresh = lowThreshold;
+    double	highThresh = highThreshold;
+
+    if (lowThresh != static_cast<int>(lowThresh) || highThresh != static_cast<int>(highThresh))
+        CV_HAL_RETURN_NOT_IMPLEMENTED("threshold with decimal values not supported");
```

Comment:
1. Looks like lowThresh and highThresh are redundant.
2. What if, lowThresh=12.999999999? I propose to use check with epsilon.

Reply:
> 1. Looks like lowThresh and highThresh are redundant.
> 2. What if, lowThresh=12.999999999? I propose to use check with epsilon.

hi @asmorkalov, updated